lib/http/http.c

```diff
@@ -559,8 +560,11 @@ gotheaders(struct http_cookie * H, uint8_t * buf, size_t buflen)
 	 */
 	if ((clen = http_findheader(H->res.headers, H->res.nheaders,
 	    "Content-Length")) != NULL) {
-		/* Parse the value (strtoull skips LWS). */
-		len = strtoull(clen, NULL, 0);
+		/* Parse the value; any base, allow trailing characters. */
```

Comment:
Why do we want to allow trailing characters here?

Reply:
Unless content-length is required to be the final header (and so far I haven't found any evidence of this) and you want to violate the robustness principle / Postel's law (i.e. bailing if somebody sends a Content-Length in the middle of headers), then we want to stop parsing at a newline.

Reply:
Nope, I'm wrong.